svr-main.c

```diff
@@ -356,6 +356,19 @@ static void main_noinetd() {
 }
 #endif /* NON_INETD_MODE */
 
+#if DROPBEAR_RESTRICT_FIXED_CLIENT_IP
+static int validate_client_ip(const char* remote_host, const char* allowed_client_ip_addr)
+{
+	return (allowed_client_ip_addr != NULL) && (strcmp(remote_host, allowed_client_ip_addr) != 0);
+}
+#else
+static int validate_client_ip(const char* remote_host, const char* allowed_client_ip_addr)
+{
+	(void)remote_host;
+	(void)allowed_client_ip_addr;
+	return 0;
```

Comment:
this turns validation always invalid, breaking the DROPBEAR_RESTRICT_FIXED_CLIENT_IP

Reply:
Didn't because of the inverse logic of validate_client_ip.
Done with new logic.